File: Data Structures & Algorithms/longest-consecutive-sequence/submission-5.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        res = 0
        st = set(nums)
        for n in nums:
            if (n - 1 in st):
                continue
            curr = n
            count = 0
            while (curr  in st):
                count += 1
                curr += 1
            res = max(res, count)
        return res

File: Data Structures & Algorithms/maximum-subarray/submission-5.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxSum = nums[0];
        int total = nums[0];
        for (int i = 1; i < nums.length; i++) {
            int n = nums[i];
            if (total < 0) {
                total = n;
            }
            else {
                total += n;
            }
            maxSum = Math.max(maxSum, total);
        }
        return maxSum;
    }
}
